### PR TITLE
feat: is_ready exits 0 when replicaSet unhealthy

### PR DESCRIPTION
**Pull request by ericboehs**
This queries the `replicas_available` for the appropriate cluster and deployment by using `$DD_ENV` and `$POD_NAME`. If the pod is unhealthy but there are too few of healthy pods in the replica set, the pod will exit 0 forcing it to accumulate a backlog. If all pods are marked as unhealthy, Traefik will drop traffic (503ing). We prefer a line to form (backlog) while new pods spin up rather than just dropping traffic.

## Testing

To test, start `bin/rails s` and run:

```
AUTOSCALING_TARGET_VALUE=0 bin/is_ready # Force an unhealthy state
echo $? # Run this each time to see exit code
AUTOSCALING_TARGET_VALUE=0 MIN_NOT_READY_PODS=10 bin/is_ready # Force a backlog to accumulate
echo $?
AUTOSCALING_TARGET_VALUE=0 MIN_NOT_READY_PODS=5 bin/is_ready # Play with min not ready to see exit code
echo $?
```

### CODE REVIEW

File: bin/is_ready [hunk 1]
```diff
@@ -9,25 +9,56 @@
...
+
+  # Extract the latest value for the given kube_deployment
+  KUBE_DEPLOYMENT=$(echo $POD_NAME | rev | cut -d'-' -f3- | rev)
+  LATEST_VALUE=$(echo "$dd_data" | tr -d '\n' | sed 's/},/\n/g' | grep "\"kube_deployment:$KUBE_DEPLOYMENT\"" | sed 's/.*pointlist":\[\[\([0-9\.]*,[0-9\.]*\)\].*/\1/' | awk -F, '{print $2}')
+
+  # Convert LATEST_VALUE to an integer
+  LATEST_VALUE=${LATEST_VALUE%.*}
+
+  # Determine if we have enough ready pods in replicaSet
+  if [ $LATEST_VALUE -le $MIN_NOT_READY_PODS ]; then
```
Comment (rmtolmach) [inline bin/is_ready:55]: When I test locally, line 55 is called out: `[: -le: unary operator expected`

File: bin/is_ready [hunk 2]
```diff
@@ -9,25 +9,56 @@
...
+    echo "Pod is unhealthy but only $LATEST_VALUE (need $MIN_NOT_READY_PODS) pods are ready for traffic. Accumulating a backlog."
+    exit 0 # Ready pods is too low, accumulate a backlog
+  else
+    echo "Pod not ready (too much traffic) and we have $LATEST_VALUE pods ready. We only need $MIN_NOT_READY_PODS."
```
Comment (rmtolmach) [inline bin/is_ready:59]: Seems like `$LATEST_VALUE` is empty. I see this message:

`Pod not ready (too much traffic) and we have  pods ready. We only need 5.`

Comment (ericboehs) [reply bin/is_ready:59]: I'm curious what $dd_data is for you. Guessing it can't find the deployment or something.

File: bin/is_ready (hunk 1 above)
Comment (ericboehs) [reply bin/is_ready:67]: I've added error checking. Try it now.

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**